File: src/main/java/com/vjiki/music/dto/CreateSongRequest.java
package com.vjiki.music.dto;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateSongRequest {
    private String title;
    private List<String> artists = new ArrayList<>();

    @JsonProperty("audio_url")
    private String audioUrl;

    private String cover;

    @JsonProperty("video_url")
    private String videoUrl;

    /**
     * SONG or SHORT (stored as plain string in DB column `songs.type`)
     */
    private String type = "SONG";

    private Boolean active = true;

    private List<CreateSongTagRequest> tags = new ArrayList<>();
}

File: src/main/kotlin/com/vjiki/music/dto/CreateSongRequest.kt
package com.vjiki.music.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class CreateSongRequest(
    val title: String,
    val artists: List<String>,
    @JsonProperty("audio_url")
    val audioUrl: String,
    val cover: String? = null,
    @JsonProperty("video_url")
    val videoUrl: String? = null,
    /**
     * SONG or SHORT (stored as plain string in DB column `songs.type`)
     */
    val type: String = "SONG",
    val active: Boolean = true,
    val tags: List<CreateSongTagRequest> = emptyList()
)

data class CreateSongTagRequest(
    val name: String,
    val weight: Double = 1.0,
    /**
     * Tag type stored in `tag.type` (e.g. GENRE / MOOD / DEFAULT)
     */
    val type: String = "DEFAULT",
    /**
     * Stored in `track_tag.source` (e.g. MANUAL / ML / IMPORT)
     */
    val source: String = "MANUAL"
)


